x/gov/msg.go

```diff
@@ -1,53 +1,82 @@
 package gov
 
 import (
-	"github.com/iov-one/weave"
 	"github.com/iov-one/weave/errors"
 	"github.com/iov-one/weave/migration"
 )
 
 const (
-	pathCreateTextProposalMsg             = "gov/create/text"
-	pathCreateElectorateUpdateProposalMsg = "gov/create/electorateUpdate"
-	pathDeleteTextProposalMsg             = "gov/delete"
-	pathVoteMsg                           = "gov/vote"
-	pathTallyMsg                          = "gov/tally"
-	pathUpdateElectorateMsg               = "gov/electorate/update"
-	pathUpdateElectionRulesMsg            = "gov/electionRules/update"
+	pathCreateProposalMsg      = "gov/create"
+	pathDeleteProposalMsg      = "gov/delete"
+	pathVoteMsg                = "gov/vote"
+	pathTallyMsg               = "gov/tally"
+	pathUpdateElectorateMsg    = "gov/electorate/update"
+	pathUpdateElectionRulesMsg = "gov/electionRules/update"
 )
 
 func init() {
-	migration.MustRegister(1, &CreateTextProposalMsg{}, migration.NoModification)
-	migration.MustRegister(1, &CreateElectorateUpdateProposalMsg{}, migration.NoModification)
+	migration.MustRegister(1, &CreateProposalMsg{}, migration.NoModification)
 	migration.MustRegister(1, &VoteMsg{}, migration.NoModification)
 	migration.MustRegister(1, &TallyMsg{}, migration.NoModification)
 	migration.MustRegister(1, &DeleteProposalMsg{}, migration.NoModification)
 	migration.MustRegister(1, &UpdateElectionRuleMsg{}, migration.NoModification)
 	migration.MustRegister(1, &UpdateElectorateMsg{}, migration.NoModification)
 }
 
-func (CreateTextProposalMsg) Path() string {
-	return pathCreateTextProposalMsg
+func (CreateProposalMsg) Path() string {
+	return pathCreateProposalMsg
 }
 
-func (m CreateTextProposalMsg) Validate() error {
-	if len(m.ElectionRuleID) == 0 {
+func (m CreateProposalMsg) Validate() error {
+	if err := m.GetMetadata().Validate(); err != nil {
+		return errors.Wrap(err, "invalid metadata")
+	}
+	if len(m.RawOption) == 0 {
+		return errors.Wrap(errors.ErrState, "missing raw options")
+	}
+	return m.Base.Validate()
+}
+
+func (m *CreateProposalBase) Validate() error {
+	if m == nil {
+		return errors.Wrap(errors.ErrInput, "missing base proposal msg info")
+	}
+
+	// Really... why not a series of if statements??
```

Comment:
@alpe loves `switch` statement :grin: 

Please either convince Alex and change it to `if` or remove the comment. :see_no_evil:

Reply:
I just fixed them.
They would all have to change to support multi-error in any case